src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -287,4 +294,108 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
 
         return new $class();
     }
+
+    /**
+     * Should this attribute be normalized?
+     *
+     * @param mixed  $object
+     * @param string $attributeName
+     * @param array  $context
+     *
+     * @return bool
+     */
+    protected function isAttributeToNormalize($object, $attributeName, &$context)
+    {
+        return !in_array($attributeName, $this->ignoredAttributes) && !$this->isMaxDepthReached(get_class($object), $attributeName, $context);
+    }
+
+    /**
+     * Sets an attribute and apply the name converter if necessary.
+     *
+     * @param array  $data
+     * @param string $attribute
+     * @param mixed  $attributeValue
+     *
+     * @return array
+     */
+    protected function setAttribute($data, $attribute, $attributeValue)
```

Comment:
I prefer to avoid references when possible (http://schlueters.de/blog/archives/125-do-not-use-php-references.html).

Typehint added.